nova/tests/unit/virt/vmwareapi/fake.py

```diff
@@ -139,17 +147,15 @@ def add_object(self, object):
         self.objects.append(object)
 
 
-def _get_object_refs(obj_type):
-    """Get object References of the type."""
-    lst_objs = []
-    for key in _db_content[obj_type]:
-        lst_objs.append(key)
-    return lst_objs
+def get_object_refs(obj_type):
+    """Get iterator over object References of the type."""
+    for obj in six.itervalues(_db_content[obj_type]):
+        yield obj.obj
 
 
-def _update_object(table, table_obj):
+def update_object(table_obj):
```

Comment:
I didn't look much into `_create_object` as it is internal to the module, and I wanted to keep the change limited to the "interface". Since it raises questions, I'll update it quickly though.